lib/version_win32.c

```diff
@@ -31,6 +31,29 @@
 #include "curl_memory.h"
 #include "memdebug.h"
 
+#if !defined(_WIN32_WINNT) || !defined(_WIN32_WINNT_WIN2K) || \
+(_WIN32_WINNT < _WIN32_WINNT_WIN2K)
+#define OLDER_THAN_WIN2K 1
+#else
+#define OLDER_THAN_WIN2K 0
+#endif
+
+/* Include the RtlGetVersion() regardless of the presence of
+   the Windows Driver Development Kit.
+*/
+#if !OLDER_THAN_WIN2K
+#include <winnt.h>
+#ifdef _MSC_VER
+#pragma comment(lib, "ntdll.lib")
```

Comment:
Thanks for working on this PR, but I guess reviewing it will be a lot easier once #7727 is settled. Not necessarily merged as it is right now as I don't think the discussion about ntdll is finished.

Reply:
I agree. That's why I made the other PR a couple of weeks ago, so there would be ample time to get that merged and then this reviewed.

Unfortunately we needed TLS 1.3 support on Windows and we couldn't wait any longer. We figured it would be better to share what we had immediately since Windows 11 will be released next week. And we didn't want other people wasting their time re-implementing the work we already did.